include/fluent-bit/flb_output.h

```diff
@@ -428,6 +432,28 @@ struct flb_output_flush {
     struct mk_list _head;              /* Link to flb_task->threads */
 };
 
+/*
+ * stores timer coros on the timer_coro_list, if the output uses them
+ */
+struct flb_output_timer_coro {
+    struct flb_config *config;         /* FLB context        */
+    struct flb_output_instance *o_ins; /* output instance    */
+    struct flb_output_coro_timer_data *timer_data; /* callback info */
+    struct flb_coro *coro;             /* parent coro addr   */
+    struct mk_list _head;              /* Link to timer_coro_list */
+};
+
+/*
+ * If the output uses timer coros, then this is used as the callback data
+ * passed to flb_sched_timer_cb_create
+ */
+struct flb_output_coro_timer_data {
```

Comment:
How about `flb_out_async_timer_cb_data` since its for the callback..

Reply:
It would be great for this function name to include the word `initialization` to differentiate it from the user callback and user provided data `void *data;`. This datastructure is explicitly used to initialize the async timer, but is not for the user provided callback. Also to differentiate from the coro cb `async_timer_cb` 

Name should align with: 
https://github.com/PettitWesley/fluent-bit/pull/27/files#r1345008308

These two functions go together - they are responsible for initializing an async_timer:
- flb_async_timer_initialization_data
- flb_async_timer_initialization_cb

Likewise we have for the user provided callback the same convention:
- cb
- data